pkg/maps/lbmap/ipv4.go

```diff
@@ -322,20 +348,37 @@ func (s *Service4Value) ToHost() ServiceValue {
 
 // +k8s:deepcopy-gen=true
 // +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
-type Backend4Key struct {
+type Backend4KeyV2 struct {
 	ID loadbalancer.BackendID
 }
 
+func NewBackend4KeyV2(id loadbalancer.BackendID) *Backend4KeyV2 {
+	return &Backend4KeyV2{ID: id}
+}
+
+func (k *Backend4KeyV2) String() string                  { return fmt.Sprintf("%d", k.ID) }
+func (k *Backend4KeyV2) GetKeyPtr() unsafe.Pointer       { return unsafe.Pointer(k) }
+func (k *Backend4KeyV2) NewValue() bpf.MapValue          { return &Backend4Value{} }
+func (k *Backend4KeyV2) Map() *bpf.Map                   { return Backend4MapV2 }
+func (k *Backend4KeyV2) SetID(id loadbalancer.BackendID) { k.ID = id }
+func (k *Backend4KeyV2) GetID() loadbalancer.BackendID   { return k.ID }
+
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type Backend4Key struct {
+	ID uint16
+}
+
 func NewBackend4Key(id loadbalancer.BackendID) *Backend4Key {
-	return &Backend4Key{ID: id}
+	return &Backend4Key{ID: uint16(id)}
 }
```

Comment:
Do we still use this function, `NewBackend4Value`, and `NewBackend4`?

Reply:
Ah good point, we no longer need these, removed.